### PR TITLE
Nightly: Fixes false positives on Policy Import

### PR DESCRIPTION
**Pull request by eloycoto**
Fix issue seen at the Nightly build 143:

https://jenkins.cilium.io/job/Cilium-Master-Nightly-Tests-All/143/consoleFull

Cilium policy version is still in old version
```
07:48:51 [Nightly] time="2018-03-15T07:48:51Z" level=info msg="CiliumPolicyAction: pod 'cilium-kgll8' still on old revision '6', need '6'" testName=NightlyK8sPolicies
07:52:21 [Nightly] time="2018-03-15T07:52:21Z" level=info msg="CiliumPolicyAction: pod 'cilium-kgll8' still on old revision '6', need '6'" testName=NightlyK8sPolicies
```

```
07:47:59 [Nightly] • Failure [984.860 seconds]
07:47:59 [Nightly] NightlyPolicies
07:47:59 [Nightly] /root/jenkins/workspace/Cilium-Master-Nightly-Tests-All/src/github.com/cilium/cilium/test/k8sT/PoliciesNightly.go:29
07:47:59 [Nightly]   PolicyEnforcement default
07:47:59 [Nightly]   /root/jenkins/workspace/Cilium-Master-Nightly-Tests-All/src/github.com/cilium/cilium/test/k8sT/PoliciesNightly.go:65
07:47:59 [Nightly]     L3:No Policy L4:No Policy L7:No Policy Destination:service [It]
07:47:59 [Nightly]     /root/jenkins/workspace/Cilium-Master-Nightly-Tests-All/src/github.com/cilium/cilium/test/k8sT/PoliciesNightly.go:70
07:47:59 [Nightly]
07:47:59 [Nightly]     cannot apply network policy for 700e0976
07:47:59 [Nightly]     Expected
07:47:59 [Nightly]         <*errors.errorString | 0xc4202a59b0>: {
07:47:59 [Nightly]             s: "Network policy cannot be imported prefix=700e0976: Timeout reached: cannot change state of resource correctly; command timed out",
07:47:59 [Nightly]         }
07:47:59 [Nightly]     to be nil
07:47:59 [Nightly]
07:47:59 [Nightly]     /root/jenkins/workspace/Cilium-Master-Nightly-Tests-All/src/github.com/cilium/cilium/test/helpers/policygen/models.go:344
07:47:59 [Nightly] ------------------------------
```

Signed-off-by: Eloy Coto <eloy.coto@gmail.com>

---
This change is [Reviewable](https://reviewable.io/reviews/cilium/cilium/3166)

### CODE REVIEW

**Comment by eloycoto** [comment]
test-nightly

**Comment by ianvernon** [comment]
@eloycoto can you please file GitHub issues that corresponds to this failure ? In case others run nightly tests in PRs and find a failure. Having issues which correlate with PRs makes tracking stuff much easier.

File: test/helpers/policygen/models.go [hunk 1]
```diff
@@ -623,8 +624,32 @@ func (t *TestSpec) NetworkPolicyApply() error {
 		helpers.KubectlApply,
 		helpers.HelperTimeout)
 
-	if err != nil {
-		return fmt.Errorf("Network policy cannot be imported prefix=%s: %s", t.Prefix, err)
+	if err == nil {
+		return nil
+	}
+
+	// Some test flakes awaiting for the policy version to being updated
```
Comment (ianvernon) [inline test/helpers/policygen/models.go:631]: ... test flakes occur while waiting for the policy version to be updated. Sometimes, pods are not scheduled on the node and the policy revision is not updated correctly.

What exactly does "policy is ok" mean? Please operationally define what being "ok" means in reference to a policy.

**Review by ianvernon — CHANGES_REQUESTED** [review]

**Comment by ianvernon** [comment]
@eloycoto is this PR still needed? It's been around for some time with no movement for months.

**Pull request closed without merging**